Drive/rover_drive_system.hpp

```diff
@@ -275,124 +230,77 @@ class RoverDriveSystem : public sjsu::common::RoverSystem
-      current_mode_ = 'T';
-    }
-    catch (const std::exception & e)
-    {
-      sjsu::LogError("Error setting translation mode!");
-      throw e;
-    }
+    HomeWheels();
+    // TODO: Find the angles close enough for an effective translation mode
+    const int left_wheel_angle  = 0;
+    const int right_wheel_angle = 60;
+    const int back_wheel_angle  = 110;
+    left_wheel_.SetSteeringAngle(left_wheel_angle);
+    right_wheel_.SetSteeringAngle(right_wheel_angle);
+    back_wheel_.SetSteeringAngle(back_wheel_angle);
+    current_mode_ = 'T';
   };
 
   /// Stops rover and updates current drive mode
   void SetExperimentalMode()
   {
-    try
-    {
-      SetWheelSpeed(kZeroSpeed);
-      current_mode_ = mc_data.drive_mode;
-    }
-    catch (const std::exception & e)
-    {
-      sjsu::LogError("Error setting translation mode!");
-      throw e;
-    }
-  }
-  double GetLeftWheelDriveAngle(double angle)
-  {
-        double left_angle = static_cast<double>(0.392 + 0.744 * x_angle + -0.0187 * pow(x_angle, 2) +
-        1.84E-04 * pow(x_angle, 3));
-        return left_angle;
-  }
-  double GetBackWheelDriveAngle(double angle)
-  {
-        return (static_cast<double>(-0.378 + -1.79 * x_angle + 0.0366 * pow(x_angle, 2) +
-        -3.24E-04 * pow(x_angle, 3)));
+    SetWheelSpeed(kZeroSpeed);
+    current_mode_ = mc_data.drive_mode;
   }
 
-  double GetOutterWheelDriveAngle(double angle)
+  int GetOutterWheelDriveAngle(int angle)
   {
     if (angle > 0)
     {
-      return static_cast<double>(0.392 + 0.744 * abs(angle) +
-                                 -0.0187 * pow(abs(angle), 2) +
-                                 1.84E-04 * pow(abs(angle), 3));
+      return 0.392 + 0.744 * abs(angle) + -0.0187 * pow(abs(angle), 2) +
+             1.84E-04 * pow(abs(angle), 3);
     }
     else
     {
-      return static_cast<double>(-(0.392 + 0.744 * abs(angle) +
-                                   -0.0187 * pow(abs(angle), 2) +
-                                   1.84E-04 * pow(abs(angle), 3)));
+      return -(0.392 + 0.744 * abs(angle) + -0.0187 * pow(abs(angle), 2) +
+               1.84E-04 * pow(abs(angle), 3));
     }
   }
 
-  double GetBackWheelDriveAngle(double angle)
+  int GetBackWheelDriveAngle(int angle)
   {
     if (angle > 0)
     {
-      return static_cast<double>(-0.378 + -1.79 * abs(angle) +
-                                 0.0366 * pow(abs(angle), 2) +
-                                 -3.24E-04 * pow(abs(angle), 3));
+      return -0.378 + -1.79 * abs(angle) + 0.0366 * pow(abs(angle), 2) +
+             -3.24E-04 * pow(abs(angle), 3);
     }
     else
     {
-      return static_cast<double>(-(-0.378 + -1.79 * abs(angle) +
-                                   0.0366 * pow(abs(angle), 2) +
-                                   -3.24E-04 * pow(abs(angle), 3)));
+      return -(-0.378 + -1.79 * abs(angle) + 0.0366 * pow(abs(angle), 2) +
+               -3.24E-04 * pow(abs(angle), 3));
```

Comment:
Does the logic still work? Will the return value just get truncated e.g. the result is 24.5918 --> return 24? Might want to write some tests to verify this.

Reply:
yes I verified this